src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.cs

```diff
@@ -180,10 +184,15 @@ private async Task ProcessQueueAsync()
                         // Mutating requests block other requests from starting to ensure an up to date snapshot is used.
                         var ranToCompletion = await work.CallbackAsync(context, cancellationToken).ConfigureAwait(false);
 
-                        // If the handling of the request failed, the exception will bubble back up to the caller, and we
-                        // request shutdown because we're in an invalid state
-                        if (!ranToCompletion)
+                        if (ranToCompletion)
+                        {
+                            // We've had a successfully mutating request, so clear out our saved state to ensure its recalculated
+                            _lastLSPSolution.Remove(context.Solution.Workspace);
```

Comment:
would it be bad to just always do this once the work completes, regardless of how it completes?

Reply:
If it doesn't complete, we shut down the server anyway, but if you think its simpler to just not have the if, I can. Won't make a lick of difference :)

Reply:
I put the if statement back. You're right, it reads nicer